File: src/java/com/puppetlabs/master/PuppetProfiler.java
package com.puppetlabs.master;

/**
 * This interface specifies Java method signatures that map to the API of the
 * Ruby Puppet Profiler.  JVM implementations of this interface can be wired in
 * to the JRuby Puppet Master and used to provide profiling support.
 */
public interface PuppetProfiler {
    Object start(String message, String[] metric_id);
    void finish(Object context, String message, String[] metric_id);
    void shutdown();
}

File: src/clj/puppetlabs/services/protocols/puppet_profiler.clj
(ns puppetlabs.services.protocols.puppet-profiler)

(defprotocol PuppetProfilerService
  "This protocol describes a service that provides a profiler that
  can be used to profile the internals of the Puppet ruby code."
  (get-profiler [this]
                "Returns an instance of `com.puppetlabs.master.PuppetProfiler` that
                can be used by the ruby Puppet profiling system."))